packages/app-desktop/gui/JoplinCloudLoginScreen.tsx

```diff
@@ -0,0 +1,121 @@
+import { CSSProperties, useEffect, useReducer, useState } from 'react';
+import ButtonBar from './ConfigScreen/ButtonBar';
+import { _ } from '@joplin/lib/locale';
+import { AppState } from '../app.reducer';
+import { clipboard } from 'electron';
+import Button, { ButtonLevel } from './Button/Button';
+const bridge = require('@electron/remote').require('./bridge').default;
+import { uuidgen } from '@joplin/lib/uuid';
+import { Dispatch } from 'redux';
+import { reducer, intitialValues, generateLoginWithUniqueLoginCode, checkIfLoginWasSuccessful } from '@joplin/lib/services/JoplinCloudLogin';
+import Setting from '@joplin/lib/models/Setting';
+
+const { connect } = require('react-redux');
+const { themeStyle } = require('@joplin/lib/theme');
+
+const loginUrl = `${Setting.value('sync.10.website')}/login`;
+const applicationsUrl = `${Setting.value('sync.10.path')}/api/applications`;
+
+interface Props {
+	themeId: string;
+	style: any;
+	dispatch: Dispatch;
+}
+
+const styles: Record<string, CSSProperties> = {
+	page: { display: 'flex', flexDirection: 'column', height: '100%' },
+	buttonsContainer: { marginBottom: '2em', display: 'flex' },
+};
```

Comment:
For styling we use RSCSS: https://joplinapp.org/help/dev/spec/desktop_styling#css-class-framework